clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/metrics/MicrometerKafkaClientMetricsSupport.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.kafka.metrics;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.kafka.KafkaClientMetrics;
+import io.micrometer.core.instrument.config.MeterFilter;
+
+/**
+ * Micrometer based implementation to provide support for registering Kafka clients from which metrics are fetched.
+ */
+public class MicrometerKafkaClientMetricsSupport implements KafkaClientMetricsSupport {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MicrometerKafkaClientMetricsSupport.class);
+
+    private final MeterRegistry meterRegistry;
+    private final Map<Producer<?, ?>, KafkaClientMetrics> producerMetricsMap = new HashMap<>();
+    private final Map<Consumer<?, ?>, KafkaClientMetrics> consumerMetricsMap = new HashMap<>();
+
+    /**
+     * Creates a new MicrometerKafkaClientMetricsSupport.
+     *
+     * @param meterRegistry The meter registry to use.
+     * @throws NullPointerException if meterRegistry is {@code null}.
+     */
+    public MicrometerKafkaClientMetricsSupport(final MeterRegistry meterRegistry) {
+        this.meterRegistry = Objects.requireNonNull(meterRegistry);
+        this.meterRegistry.config().meterFilter(MeterFilter.denyNameStartsWith("kafka.producer.waiting.threads"));
```

Comment:
By default, the Micrometer `KafkaMetrics` class adds quite many meters for each client. Above is a way to reduce this amount. We might add further such entries in the future.
(Also, the Micrometer `KafkaMetrics` class might exclude some metrics itself in an upcoming version, as indicated in
https://github.com/micrometer-metrics/micrometer/issues/2628#issuecomment-917096676.)

Reply:
As there are really many metrics reported here (>190), I think it would be good to define a configuration property to configure a whitelist for the metrics to be actually reported. I'll look into updating this PR accordingly...

Reply:
I've now amended the commit to only use a subset of Kafka client metrics by default.
Via added config properties, this set of metrics can either be replaced by an own set of metrics (defined via metrics prefixes), or additional metrics can be defined on top of the default metrics.